packages/core/saved-objects/core-saved-objects-migration-server-internal/src/zdt/model/stages/create_target_index.ts

```diff
@@ -48,10 +47,11 @@ export const createTargetIndex: ModelStage<'CREATE_TARGET_INDEX', 'UPDATE_ALIASE
 
   return {
     ...state,
-    controlState: 'UPDATE_ALIASES',
+    controlState: aliasActions.length ? 'UPDATE_ALIASES' : 'INDEX_STATE_UPDATE_DONE',
```

Comment:
If we create a new index then we will likely always have to create the aliases, yeah.

> But I wonder why we add aliases as a second step instead of creating the index with the aliases already

That's a good one, I added it as the list of optimizations to https://github.com/elastic/kibana/issues/150309